### PR TITLE
macOS say support

### PR DESCRIPTION
**Pull request by klaas**

### CODE REVIEW

File: lib/tts-providers/macOs.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+const crypto = require('crypto');
+const fs = require('fs');
+const http = require('http');
+const path = require('path');
+const fileDuration = require('../helpers/file-duration');
+const settings = require('../../settings');
+const logger = require('sonos-discovery/lib/helpers/logger');
+var exec = require('child_process').exec;
+
+function mac_os_say(phrase, voice) {
```
Comment (jishi) [inline lib/tts-providers/macOs.js]: use camelCase function names

File: lib/tts-providers/macOs.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          duration,
+          uri: expectedUri
+        };
+      });
+  } catch (err) {
+    logger.info(`announce file for phrase "${phrase}" does not seem to exist, downloading`);
+  }
+
+ return new Promise((resolve, reject) => {
+    exec('say -v ' + voice + ' "' + phrase + '" -o ' + filepath,
```
Comment (jishi) [inline lib/tts-providers/macOs.js]: use string templating, i.e `` `say -v ${voice} "${phrase}" -o ${filepath}` ``

**Comment by jishi** [comment]
How would you configure it to use the macOS TTS provider?

**Comment by klaas** [comment]
There is not really a need to configure the macOS TTS provider, because the say command works without any configuration and then uses the system voice from System Preferences.

My suggestion would be to configure a default voice in the settings.json of node-sonos-http-api. This would triggers the use of the macOS TTS. Please see the attached changes.

(Yesterdays commit was more or less by accident :-))

**Comment by jishi** [comment]
Weel, I get that you don't need any configuration per se, but the approach I have taken is that it will choose provider based on what you configured (in a certain priority order) as a mean to simplify it. 

We could let voice be configurable in settings.json, and let that be the trigger to prefer macos TTS over google.

**Comment by klaas** [comment]
I did as you suggested to choose the voice itself. The only other option is the "speech rate". 

I'm not sure what format you prefer for the configuration inside the `settings.json`. Should I make it like this:

```
"macSay" : {
	"voice": "Alex",
	"rate": "30"
}
```

If I look at the keys for the different TTS providers, maybe you prefer "apple" over "macSay"?

**Comment by jishi** [comment]
Doesn't really matter, still boils down to the documentation to figure out what configurable options you have. `macOS` or `macSay` seems like plausible options to me.

**Comment by klaas** [comment]
Hi jishi, I just updated the code and the README.md to support the "rate" setting. Anything more that I can do?

**Comment by jishi** [comment]
No, looks great. I just want to test it out before I merge it :) Need to bring my macbook home to verify, will do this weekend.

**Comment by klaas** [comment]
Sounds great!

**Comment by jishi** [comment]
Rebased and merged so it lost track of this PR, closing this.

**Pull request closed without merging**